### ISSUE TITLE
Rule `jsdoc/no-undefined-types` reports the type `IterableIterator` is undefined

### ISSUE THREAD

**Issue by jaydenseric** [gajus/eslint-plugin-jsdoc#1066]
## Expected behavior

The following:

```js
// @ts-check

const messages = [];

for (const [index, message] of /** @type {IterableIterator<[number, unknown]>} */ (messages.entries())) {
  // …
}
```

Should not result in a rule `jsdoc/no-undefined-types` error.

## Actual behavior

It results in the error:

```
The type 'IterableIterator' is undefined.
```

## ESLint Config

```json
{
  "extends": "plugin:jsdoc/recommended-typescript-error",
  "settings": {
    "jsdoc": {
      "mode": "typescript"
    }
  },
}
```

## ESLint sample

```js
// @ts-check

const messages = [];

for (const [index, message] of /** @type {IterableIterator<[number, unknown]>} */ (messages.entries())) {
  // …
}
```

## Environment

- Node version: v20.0.0
- ESLint version v8.39.0
- `eslint-plugin-jsdoc` version: v43.1.1

**Comment by brettz9** [comment]
Closing as a dupe of #888 though feel free to comment further as needed.

**Comment by jaydenseric** [comment]
Why does the config `plugin:jsdoc/recommended-typescript-error` enable the `jsdoc/no-undefined-types` rule, when TypeScript itself reports errors for invalid JSDoc types? Is an easy fix for this issue to turn it off in the recommended TypeScript config, and regarding https://github.com/gajus/eslint-plugin-jsdoc/issues/888, to make the rule `jsdoc/no-undefined-types` a no-op in TypeScript mode?

**Comment by brettz9** [comment]
@jaydenseric : SGTM. How about you re-post your comment in #888 so others may chime in?
